src/appleseed/foundation/math/fresnel.h

```diff
@@ -383,6 +434,138 @@ void normal_reflectance_dielectric(
     normal_reflectance *= normal_reflectance;
 }
 
+template <typename SpectrumType, typename T>
+inline void fresnel_reflectance_conductor(
+    SpectrumType&           reflectance,
+    const SpectrumType&     n,
+    const SpectrumType&     k,
+    const T                 cos_theta_i)
+{
+    //
+    //      (n^2 + k^2) cos_theta^2 - 2n cos_theta + 1
+    // Rp = ------------------------------------------
+    //      (n^2 + k^2) cos_theta^2 + 2n cos_theta + 1
+    //
+
+    //
+    //      (n^2 + k^2) - 2n cos_theta + cos_theta^2
+    // Rs = ----------------------------------------
+    //      (n^2 + k^2) + 2n cos_theta + cos_theta^2
+    //
+
+    //               Rp + Rs
+    // reflectance = -------
+    //                  2
+
+    typedef typename impl::GetValueType<SpectrumType>::ValueType ValueType;
+
+    const T cos_theta = std::abs(cos_theta_i);
+    const T cos_theta2 = square(cos_theta);
+
+    const SpectrumType n2 = n * n;
```

Comment:
Not sure how this compares in terms of generated code to:

    SpectrumType n2(n);
    n2 *= n2;